### DISCUSSION TITLE
6.3 Visualizing the results for individual pairs plotting error

### DISCUSSION THREAD

**Discussion by yabeykoon** [Katsevich-Lab/sceptre#145] [Q&A] [answered]
I'm trying to include the argument as in your example in part 6.3 of SCEPTRE book:
```
`disc_pair <- [plot_response_grna_target_pair](https://rdrr.io/pkg/sceptre/man/plot_response_grna_target_pair.html)(sceptre_object = sceptre_object_lowmoi,
                                            response_id = "ACADVL",
                                            grna_target = "STAT3")
[plot_grid](https://wilkelab.org/cowplot/reference/plot_grid.html)(pc_pair, disc_pair)`
```
This is adjusted code for my own high MOI perturb-seq experiment:
```
disc_pair <- plot_response_grna_target_pair(sceptre_object = sceptre_object,
                                            response_id = "ENSG00000134121",
                                            grna_target = "candidate_enh_9")
plot_grid(disc_pair)

```

But I'm getting the error:
```
`Error in plot_response_grna_target_pair(sceptre_object = sceptre_object,  :
  gRNA target `candidate_enh_9` is not present within the data.
Execution halted
`
```
I've run this pipeline to completion before and I know there's an element called `candidate_enh_9` in my grna_target column because from the previous run when I output :
```
result_optimization18_sceptre_combined <- get_result(
  sceptre_object = sceptre_object,
  analysis = "run_discovery_analysis"
)
```
I get the data frame as with candidate_enh_9 populating the grna_target column. Screenshot attached here. How can I generate the plots comparing the treatment vs control pairs?
<img width="707" alt="Screenshot 2024-09-16 at 1 29 03 PM" src="https://github.com/user-attachments/assets/da6b5d95-541c-4dd7-ae91-01b6a45be97a">

**Comment by yabeykoon** [comment]
After playing around for a while, I was able to get this to work by plugging in the contents in the grna_id from my results table for the grna_target variable in the argument. My contents in grna_id vs grna_target are not the same whereas in your example, they have the same content. I have information for individual sgRNA in grna_id column and something like "candidate_enh_x" in my grna_target column. Then for generating the p_value, I had to use actual grna_target cell content with response_id as a separate argument added to your example code.

**Reply by timothy-barry** [reply]
Hello,

Thanks for the discussion. I am not sure I totally understand, but as an initial question, what is your `grna_integration_strategy` set to?

Tim

**Comment by timothy-barry** [comment, marked as answer]
Hi Yash, I pushed an update to the package that I believe fixes `plot_response_grna_target_pair()` when `grna_integration_strategy` is set to `"singleton"`. Would you mind redownloading the package and verifying? Thanks!
